idea/src/org/jetbrains/kotlin/idea/intentions/JoinDeclarationAndAssignmentIntention.kt

```diff
@@ -140,6 +142,10 @@ class JoinDeclarationAndAssignmentIntention : SelfTargetingRangeIntention<KtProp
 
         val firstAssignment = assignments.firstOrNull() ?: return null
         if (assignments.any { it !== firstAssignment && it.parent.parent is KtSecondaryConstructor }) return null
+        if (!property.isLocal &&
+            firstAssignment.parent != propertyContainer &&
+            firstAssignment.right.safeAs<KtNameReferenceExpression>()?.mainReference?.resolve() is KtParameter
```

Comment:
This check doesn't work if the initializer is an expression that uses a parameter:
```kotlin
class Id {
    val id: Int<caret>

    constructor(id: Int) {
        this.id = id + 1
    }

    constructor() : this(0)
} 
```

Reply:
Fixed in 7d4ff28c96c48aafb22735427e441d2688090681